credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt

```diff
@@ -234,9 +255,14 @@ class PresentationExchangeTest {
       )
       val vcJwt = vc.sign(issuerDid)
 
+      assertThrows<PresentationExchangeException> {
+        PresentationExchange.satisfiesPresentationDefinition(listOf(vcJwt), pd)
+      }
```

Comment:
Curious how this compares to the test below that calls `assertFailure`? Is the difference that the type is asserted vs the message?

Reply:
agh. yea, i don't love the way i wrote this! really what i wanted was something like this, but i couldn't quite get it to work. so i'm wanting to test two different things - 1) that a specific exception was thrown, and 2) that it fails with a specific message

```
assertFailsWith< PresentationExchangeException> {
        PresentationExchange.satisfiesPresentationDefinition(listOf(vcJwt), pd)
      }.messageContains("Failed to parse VC payload as JSON.")
```

Reply:
Gotcha! Perhaps it's worth making our own test assert file which would include this `assertFailsWith` function initially